lib/pact_broker/api/decorators/matrix_decorator.rb

```diff
@@ -111,8 +111,8 @@ def branches(branch_versions, base_url)
         end
 
         def environments(deployed_versions, released_versions, base_url)
-          (deployed_versions + released_versions).sort_by(&:created_at).collect(&:environment).uniq.collect do | environment |
-            environment_decorator_class.new(environment).to_hash(user_options: { base_url: base_url })
+          (deployed_versions + released_versions).sort_by(&:created_at).collect { |v| [v.environment, v.is_a?(Deployments::DeployedVersion) ? v.application_instance : nil] }.uniq.collect do | environment, application_instance |
```

Comment:
do we need to consider if its a DeployedVersion and the application_instance is unset or will it always be present but nil, in `Deployments::DeployedVersion`

Reply:
if record is of type ReleasedVersion then we would get error cause ReleasedVersion table doesn't have a column for storing app instance

Reply:
in case deployed verion type the app instance value is going to be present. The value can be null as well. The column app instance is going to exists